pkg/apis/sources/v1alpha1/rabbitmq_validation.go

```diff
@@ -33,6 +33,21 @@ func (current *RabbitmqSource) Validate(ctx context.Context) *apis.FieldError {
 		}
 	}
 
+	var errs *apis.FieldError
+	if current.Spec.RabbitmqClusterReference == nil {
+		return apis.ErrMissingField("rabbitmqClusterReference").ViaField("spec")
+	} else {
+		if current.Spec.RabbitmqClusterReference.Name == "" {
+			return errs.Also(apis.ErrMissingField("name").ViaField("rabbitmqClusterReference").ViaField("spec"))
+		}
+		if current.Spec.RabbitmqClusterReference.Namespace == "" {
+			return errs.Also(apis.ErrMissingField("namespace").ViaField("rabbitmqClusterReference").ViaField("spec"))
+		}
+		if current.Spec.RabbitmqClusterReference.ConnectionSecret != nil {
+			return apis.ErrDisallowedFields("connectionSecret").ViaField("rabbitmqClusterReference").ViaField("spec")
```

Comment:
Might be copy pasta issue with the RabbitmqBrokerConfig, but for sources `connectionSecret` is an allowed field

Reply:
answered on the other comment